components/channel_move_to_sub_menu/index.tsx

```diff
@@ -0,0 +1,172 @@
+            {
+                id: 'ChannelMenu-moveToDivider',
+                text: (<span className='MenuGroup menu-divider'/>),
+            },
+            {
+                id: `moveToNewCategory-${props.channel.id}`,
+                icon: (<FolderMoveOutlineIcon size={16}/>),
+                direction: 'right' as any,
+                text: formatMessage({id: 'sidebar_left.sidebar_channel_menu.moveToNewCategory', defaultMessage: 'New Category'}),
+                action: handleMoveToNewCategory,
+            },
+        ];
+
+        return [...allCategories, ...dividerAndNewCategory];
+    }
+
+    function filterCategoriesBasedOnChannelType(categories: ChannelCategory[], isDmOrGm = false) {
+        if (isDmOrGm) {
+            return categories.filter((category) => category.type !== CategoryTypes.CHANNELS);
+        }
+
+        return categories.filter((category) => category.type !== CategoryTypes.DIRECT_MESSAGES);
+    }
+
+    function getMoveToCategorySubmenuItems(categories: ChannelCategory[]) {
+        const isSubmenuOneOfSelectedChannels = multiSelectedChannelIds.includes(props.channel.id);
+
+        // If sub menu is in channel header dropdown OR If multiple channels are selected but the menu is open outside of those selected channels
+        if (props.inHeaderDropdown || !isSubmenuOneOfSelectedChannels) {
+            const isDmOrGm = props.channel.type === Constants.DM_CHANNEL || props.channel.type === Constants.GM_CHANNEL;
+            const filteredCategories = filterCategoriesBasedOnChannelType(categories, isDmOrGm);
+            return createSubmenuItemsForCategoryArray(filteredCategories);
+        }
+
+        const areAllSelectedChannelsDMorGM = multiSelectedChannelIds.every((channelId) => allChannels[channelId].type === Constants.DM_CHANNEL || allChannels[channelId].type === Constants.GM_CHANNEL);
+        if (areAllSelectedChannelsDMorGM) {
+            const filteredCategories = filterCategoriesBasedOnChannelType(categories, true);
+            return createSubmenuItemsForCategoryArray(filteredCategories);
+        }
+
+        const areAllSelectedChannelsAreNotDMorGM = multiSelectedChannelIds.every((channelId) => allChannels[channelId].type !== Constants.DM_CHANNEL && allChannels[channelId].type !== Constants.GM_CHANNEL);
+        if (areAllSelectedChannelsAreNotDMorGM) {
+            const filteredCategories = filterCategoriesBasedOnChannelType(categories, false);
+            return createSubmenuItemsForCategoryArray(filteredCategories);
+        }
+
+        // If we have a mix of channel types, we need to filter out both the DM and Channel categories
+        const filteredCategories = categories.filter((category) => category.type !== CategoryTypes.CHANNELS && category.type !== CategoryTypes.DIRECT_MESSAGES);
+        return createSubmenuItemsForCategoryArray(filteredCategories);
+    }
+
+    if (!categories) {
+        return null;
+    }
+
+    return (
+        <React.Fragment>
```

Comment:
The outer fragment isn't needed by the look of it

Reply:
removed the outer fragment